base/src/main/scala/clazz/Applicative.scala

```diff
@@ -2,8 +2,16 @@ package scato
 package clazz
 
 abstract class Applicative[F[_]] {
-  def apply: Apply[F]
+
   def pure[A](a: A): F[A]
+  def ap[A, B](fa: F[A])(f: F[A => B]): F[B]
+
+  def map[A, B](ma: F[A])(f: (A) => B): F[B] = ap(ma)(pure(f))
+
+  final val apply: Apply[F] = new Apply[F] {
```

Comment:
@jbgi what if a write a newtype `Foo`, which implement a class `Bar` (for which `Apply` is a superclass)... and now I want to define an `Applicative` for `Foo`. 

With the approach you suggest, it seems to me that I would be forced to first define the `Apply` in my `Applicative` and then refer to the applicative instance in my `Bar` which is bad, because that introduce a dependency between my `Bar` instance and `Applicative`.

Instead I would rather follow an approach like I used to define `Free`, here:
https://github.com/aloiscochard/scato/blob/master/free/src/main/scala/monad/FreeInstances.scala

(Let's pretend `freeBindRec` is implemented, and `freeBind` is shared to both).

I have the feeling the approach I suggest (leaving this field non final), is extremely important to support this sort of use-case.

What do you think?

(Note: `lazy` val not free, which use them carefully, I think the cost of instantiation outweigh the lazy val cost in those context, but yeah... benchmark would show if I'm right).